### ISSUE TITLE
prefer_interpolation_to_compose_strings and raw strings

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/linter#2490]
```dart
main() {
  final String foo = 'Hello';
  final String bar = foo + r' /world\';
  print(bar);
}
```

This triggers the prefer_interpolation_to_compose_strings lint, however that doesn't seem to be appropriate as I am not aware of how to do string interpolation with raw strings?

**Comment by jonahwilliams** [comment]
You can use adjacent string literals for that `final String bar = 'Hello' r' /world\';`

**Comment by goderbauer** [comment]
I guess, that works. But it doesn't read as cleanly:

```dart
final String bar = '$foo' r' /world\';
```

vs

```dart
final String bar = foo + r' /world\';
```

**Comment by pq** [comment]
Yeah, my sense is we should probably skip cases concatenating raw strings.

**Comment by bwilkerson** [comment]
At the very least we should not be suggesting string interpolation as the solution.

**Comment by lrhn** [comment]
You can do `"$foo${r' /world\'}"` too. Not sure whether better or worse.
